Yoast/wordpress-seo
# Dependency injection

## Summary

This PR can be summarized in the following changelog entry:

* No user-facing changes

## Relevant technical choices:

* Introduces a dependency injection container based on symfony/dependency-injection.
* Introduces a new Loader class that is automatically injected with all classes matching the Integration interface.
* Introduces Conditionals as a way to determine which integrations should be loaded.
* Refactors the src directory to ensure all classes are loaded through the dependency injection container where possible.

## Test instructions

This PR can be tested by following these steps:

* Run a site on PHP 5.2. It should not crash and nothing in the src directory should be loaded.
* Run a site on PHP 5.6+ it should not crash, the Loader should be getting run but no indexable related migrations or code should be run.
* Run a site on PHP 5/6+ with the `YOAST_SEO_INDEXABLES` feature flag on. It should not crash and when saving a post, term or author it should be added to the indexables table.

## Documentation

## Quality assurance


Fixes #13079
